### ISSUE TITLE
关于线程池这一章的问题

### ISSUE THREAD

**Issue by Chen-Xuming** [downdemo/Cpp-Concurrency-in-Action-2ed#3]
**“往线程池添加任务会增加任务队列的竞争，lock-free 队列可以避免这点但存在乒乓缓存的问题。为此需要把任务队列拆分为线程独立的本地队列和全局队列，当线程队列无任务时就去全局队列取任务”**

在这一小节中，submit函数中的local_queue永远是nullptr，原因如下：
local_queue指针在worker_thread中赋值，woker_thread运行在子线程，而submit函数是在main线程中调用的，二者不是在同一线程，所以二者的local_queue根本不是同一个。

所以这段代码无法达到想要的目的。

**Comment by downdemo** [comment]
You're right. This can be done by hashing the thread ID to access each thread's queue. I'll fix this at my leisure.
